src/ArmTemplates/Common/API/Clients/Abstractions/ApiClientBase.cs

```diff
@@ -23,7 +23,7 @@ public abstract class ApiClientBase
 
         protected string BaseUrl { get; private set; } = GlobalConstants.BaseManagementAzureUrl;
 
-        protected AzureCliAuthenticator Auth { get; private set; } = new AzureCliAuthenticator();
+        protected virtual AzureCliAuthenticator Auth { get; private set; } = new AzureCliAuthenticator();
```

Comment:
Personally, I would prefer another implementation. Negatives of current approaches that I see:
- you have virtual only for mocking purposes
- you have **very** error-prone approach with having a string as a method name in your mock setup. Imagine another developer changes name of `CallApiManagementAsync`, and tests will just break with some mock-could-not-be-found **runtime** exception, which is always unclear.

As we discussed before, we could make another interface for `httpClient` and for `Authenticator`, but it requires a lot of unnecessary work + there is no point in having a separate interface for HttpClient, because we have a single-implementation here.

But looking at code I see another great approach for handling that - using registration in DI for httpclients: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-6.0

I like this approach because:
- you will make a constructor for `ApiClientBase`, which get a `HttpClient`. And in test you can create your own httpClient, which will return everything you want, so it is possible to test such code
- we will fix the non-pooled and non-disposed `HttpClients`: you can see we are creating a separate HttpClient for every single client used in program, so I guess at some moment, if extractor runs long enough, or the connection-pool will be exhausted, we will get troubles. Since we have IServiceCollection, we can try to use `IHttpClientFactory`, which uses pooled HttpClients and controls reusing of connections + disposing them.

What do you think?

Reply:
As was discussed, https://github.com/Azure/azure-api-management-devops-resource-kit/pull/764 PR is merged hence we can resolve the conversation, having that httpClientFactory is added and used in the Client via DI.